File: week1/임성후/[leetcode] find pivot index.go
package main

import "fmt"

func main() {
	nums := []int{-1, -1, 0, 1, 1, 0}
	u := pivotIndex(nums)
	fmt.Println(u)
}

func pivotIndex(nums []int) int {
	if len(nums) == 1 {
		return 0
	}
	right_sum := getSum(nums[1:])
	left_sum := 0

	for i := 0; i < len(nums)-1; i++ {
		if left_sum == right_sum {
			return i
		} else {
			left_sum += nums[i]
			right_sum -= nums[i+1]
		}
	}

	if left_sum == right_sum {
		return len(nums) - 1
	}

	return -1
}

func getSum(nums []int) int {
	sum := 0

	for _, val := range nums {
		sum += val
	}
	return sum
}

File: week1/임성후/[leetcode] find pivot index.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        left = 0
        right = len(nums)
        for m in range(len(nums)):
            if sum(nums[left: m]) == sum(nums[m+1: right]):
                return m
        else:
            return -1


s = Solution()
print(s.pivotIndex([1, 7, 3, 6, 5, 6]))
print(s.pivotIndex([1, 2, 3]))
